### PR TITLE
Add docs for VisibilityStateEntry

### PR DESCRIPTION
**Pull request by chrisdavidmills**
### Description

Chrome 115 adds the `VisibilityStateEntry` class, which represents a performance entry of type “visibility-state”. See https://chromestatus.com/feature/5683502144028672.

This PR adds documentation for the new class, and updates surrounding Performance API docs to mention it as appropriate.

See also my [research doc](https://docs.google.com/document/d/1nIsWNlAdkwvzlqcNJ0aC0FTAxN8IDcYmYO42kUwhGvM/edit) for more information on exactly what work has been done.

### CODE REVIEW

File: files/en-us/web/api/visibilitystateentry/index.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+page-type: web-api-interface
+status:
+  - experimental
+browser-compat: api.VisibilityStateEntry
+---
+
+{{APIRef("Performance API")}}{{seecompattable}}
+
+The **`VisibilityStateEntry`** interface provides timings of page visibility state changes, i.e., when a tab changes from the foreground to the background or vice versa.
+
```
Comment (noamr) [inline files/en-us/web/api/visibilitystateentry/index.md:13]: An important bit that I'm missing here is that the document would always have at least one `visibility-state` entry at time 0 - which would represent the visiblity state when the document was created.

File: files/en-us/web/api/visibilitystateentry/index.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  return visibilityStateEntries.some(
+    (e) => e.startTime < fcpEntry.startTime && e.name === "hidden"
+  );
+}
+```
+
+## Specifications
+
+{{Specifications}}
+
```
Comment (noamr) [inline files/en-us/web/api/visibilitystateentry/index.md:76]: Would be good to have a link to the spec:
https://html.spec.whatwg.org/multipage/interaction.html#visibilitystateentry

Comment (Mawajingga) [reply files/en-us/web/api/visibilitystateentry/index.md:76]: Delete

File: files/en-us/web/api/visibilitystateentry/index.md (hunk 1 above)
Comment (Mawajingga) [reply files/en-us/web/api/visibilitystateentry/index.md:13]: Delete

**Review by noamr — COMMENTED** [review]
LGTM

**Comment by noamr** [comment]
> LGTM

(I mean a non-authoritative LGTM from a technical point of view)

**Comment by chrisdavidmills** [comment]
> > LGTM
> 
> (I mean a non-authoritative LGTM from a technical point of view)

Yup, that's perfect. Thx for taking the time to review!

**Review by Elchi3 — COMMENTED** [review]
Could you also update https://developer.mozilla.org/en-US/docs/Web/API/performance_property to say it is Window-context only?

Also, I wrote a guide about "Performance data" a while ago and it would be nice to add this to it: https://developer.mozilla.org/en-US/docs/Web/API/Performance_API/Performance_data 
There is a section on this page about buffer sizes and my source is https://w3c.github.io/timing-entrytypes-registry/#registry. However, I don't see `visibility-state` performance entries in that registry. Maybe @noamr knows more.

**Comment by Elchi3** [comment]
Also, I wonder if we should integrate this into https://developer.mozilla.org/en-US/docs/Web/API/Performance_API/Navigation_timing

**Review by Elchi3 — COMMENTED** [review]
Also, updates to PerformanceEntry would be good:

- https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/duration add "visibility-state" to the list "For the following entry types, duration is not applicable, and in this case the value is always 0:"
- https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/name add to the table
- https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/startTime add to "Value" section

**Comment by chrisdavidmills** [comment]
Thanks for the review @Elchi3 . To summarize the changes in my next commit (it weirdly won't let me add replies to your individual comments):

* I have added `VisibilityStateEntry` into the table on the performance_property page
* I have added it in all appropriate places in the Performance_data page. I couldn't see a max buffer entries defined for it either, so I've entered it as `infinite` for now. But yes, maybe @noamr can help confirm this value.
* I've not added anything to the Navigation_timing page, as I'm not really sure what you were thinking of here. It doesn't really seem directly relevant to me.
* Changes to `name`, `duration`, and `startTime` — all done.

**Comment by noamr** [comment]
> Thanks for the review @Elchi3 . To summarize the changes in my next commit (it weirdly won't let me add replies to your individual comments):
> 
> * I have added `VisibilityStateEntry` into the table on the performance_property page
> * I have added it in all appropriate places in the Performance_data page. I couldn't see a max buffer entries defined for it either, so I've entered it as `infinite` for now. But yes, maybe @noamr can help confirm this value.

It's 50. I should add it to the registry spec.

> * I've not added anything to the Navigation_timing page, as I'm not really sure what you were thinking of here. It doesn't really seem directly relevant to me.
> * Changes to `name`, `duration`, and `startTime` — all done.

**Comment by chrisdavidmills** [comment]
> It's 50. I should add it to the registry spec.

Thanks @noamr ; page updated.

**Review by Elchi3 — APPROVED** [review]
Great work, thanks @chrisdavidmills 👍

**Pull request merged**